### PR TITLE
12.0 mig base sms client

### PR DESCRIPTION
**Pull request by cvinh**

### CODE REVIEW

**Comment by cvinh** [comment]
@sbejaoui we did not see that you were working on the migration of this module... our bad..
By the way we found 2 issues that we fixed in that PR. Could you have a look and include then in your PR ?

**Comment by sbejaoui** [comment]
Hi @cvinh ,

I will consider the fixes in my PR. From my side, I made some fixes too:
- Remove usless keychain dependency
- remove explicit cr.commit
- set sms company based on user company

I am working on some improvement of the module:
- Properly define sms templates
- Integrate sms templates in ir.cron

File: base_sms_client/security/ir.rule.csv [hunk 1]
```diff
@@ -1,4 +1,3 @@
 id,name,model_id:id,groups:id,perm_read,perm_write,perm_create,perm_unlink,domain_force
 edit_access_sms_gateway,Edit Access Sms Gateway,model_sms_gateway,,False,True,True,True,"[('company_id','=', user.company_id.id)]"
 read_access_sms_gateway,Read Access Sms Gateway,model_sms_gateway,,True,False,False,False,"['|', '|', ('company_id','=', user.company_id.id),('company_id','child_of',[user.company_id.id]),('company_id','=',user.company_id.parent_id.id)]"
-edit_access_sms,Edit Access Sms,model_sms_sms,,True,True,True,True,"[('company_id','=', user.company_id.id)]"
```
Comment (sbejaoui) [inline base_sms_client/security/ir.rule.csv:4]: I fixed this by setting the sms company based on the create user company

File: base_sms_client/models/sms_sms.py [hunk 2]
```diff
@@ -69,7 +69,8 @@ def _check_gateway_method(self):
     @api.multi
     def _check_gateway_permission(self):
         self.ensure_one()
-        if self.gateway_id._check_permissions():
+        user = self.create_uid
+        if self.gateway_id._check_permissions(user):
```
Comment (sbejaoui) [inline base_sms_client/models/sms_sms.py:73]: changing the method param can break compatibility in other modules (OCA or custom). I propose to fix this like:

```suggestion
        if self.sudo(self.create_uid.id).gateway_id._check_permissions():
```

**Comment by sbejaoui** [comment]
I added your fix to #209 , we can close this one and follow-up with me on the other one

**Pull request closed without merging**